delegation/src/main/java/com/iluwatar/delegation/simple/PrinterController.java

```diff
@@ -0,0 +1,21 @@
+package com.iluwatar.delegation.simple;
+
+public class PrinterController extends AbstractPrinterController {
+
+  public PrinterController(Printer printer) {
```

Comment:
In my opinion there is one layer too many of abstraction. I would remove `AbstractPrinterController` and do everything in `PrinterController`.